### DISCUSSION TITLE
Google Cloud Functions Webhook with PyWA

### DISCUSSION THREAD

**Discussion by ecnjava** [david-lev/pywa#89] [Q&A]
Hi guys!! 
I'm using GCF webhook and PyWA.... 
Throughout GC Logs explorer, I can see the incoming messages accordingly, but my Flask server is not being triggered whenever a message arrives (@wa_ret.on_message(filters.command("start")))..... Could you please advise me??? 

Here is my source code: 

import flask  # pip3 install flask
from pywa import WhatsApp, types

flask_app = flask.Flask(__name__)

async def init_wa():
    global wa_ret
    if wa_ret is None:
        wa_ret = WhatsApp(
            phone_id="3",
            server=flask_app,
            token="1",
            verify_token='1',
            api_version=21.0,
            continue_handling=True)

        @wa_ret.on_message(filters.command("start"))
        async def on_start(_: WhatsApp, m: Message):
            await m.reply("Who are you?")

@https_fn.on_request()
def webhook(request: https_fn.Request) -> https_fn.Response:
    if request.method == 'GET':
        mode = request.args.get('hub.mode')
        token = request.args.get('hub.verify_token')
        challenge = request.args.get('hub.challenge')
        challenge_200 = challenge, 200

        if mode and token:
            if mode == 'subscribe' and token == '1':
                return challenge_200
            else:
                return 'Invalid verification token', 403
        else:
            return 'Missing parameters', 400

    elif request.method == 'POST':
        if wa_ret is None:
            asyncio.run(main())  # Initialize WhatsApp if not done

    return jsonify({'status': 'success'})

**Comment by felipe7710** [comment]
@ecnjava Did you register your URL webhook in Meta Portal ?
